### DISCUSSION TITLE
text_box()

### DISCUSSION THREAD

**Discussion by Alex-Jordan** [openwebwork/pg#737] [Ideas]
I've been thinking that there are pedagogical considerations where sometimes, you want to remove access to answer input assistance like MathQuill or MathView (if it's provided in the first place). You might want them in general for your course, but not fora  particular exercise. Furthermore, it may not be a matter of instructor preference, but rather the intent of that exercise really is fundamentally about entering something in "calculator keyboard notation". 

In the few cases this comes up at my school, the question is something like "how can you type this into a calculator or CAS?" And it's an expression like `\frac{x+1}{x+2}`. So it wants something like `(x+1)/(x+2)`, checking that students don't do things like `x+1/x+2`. It messes with the intent of the exercise when MathQuill for example auto-formats the student's answer. And yet you don't want to disable MQ for the whole course.

So I'm wondering about a new type of answer blank. It would be like ans_rule() in most ways. But it would also behave like essay_box() in the sense that when a user enters that field, there is no answer entry assistance. I've been abusing essay_box() for this purpose. 

Does anyone have thoughts about this? I could continue abusing essay_box(). However if it would be accepted, I would write a new thing in PGbasicmacros.pl that maybe I'd call text_box(). I'm having trouble convincing myself that it's OK to write a flavor of answer input field where the purpose is to override things like MathQuill. It feels like MQ is a webwork2 feature about how a user interacts with exercises, and PG should be confined to pedagogy of individual questions. But there's a muddy overlap here.

**Comment by drgrice1** [comment]
For MathQuill you can already do this by passing `mathQuillOpts => 'disabled'` to the `cmp` method.  The `parserWordCompletion.pl` macro sets this for example.  See line 130 of that macro.

**Reply by Alex-Jordan** [reply]
Oh, this will help a lot with my local situation. I was unaware.

MQ is naturally the dominant tool but a few still prefer MathView....
